dynamo/tools/clustering.py

```diff
@@ -174,36 +173,61 @@ def hdbscan(
 
 
 def leiden(
-    adata,
-    use_weight=True,
-    weight=None,
-    initial_membership=None,
-    adj_matrix=None,
-    adj_matrix_key=None,
-    result_key=None,
-    layer=None,
-    obsm_key=None,
-    selected_cluster_subset: list = None,
-    selected_cell_subset=None,
-    directed=False,
-    copy=False,
+    adata: AnnData,
+    resolution: float = 1.0,
+    use_weight: bool = True,
+    weight: Optional[list] = None,
+    initial_membership: Optional[list] = None,
+    adj_matrix: Optional[csr_matrix] = None,
+    adj_matrix_key: Optional[str] = None,
+    result_key: Optional[str] = None,
+    layer: Optional[str] = None,
+    obsm_key: Optional[str] = None,
+    selected_cluster_subset: Optional[list] = None,
+    selected_cell_subset: Optional[list] = None,
+    directed: bool = False,
+    copy: bool = False,
     **kwargs
 ) -> anndata.AnnData:
     """Apply leiden clustering to adata.
-    For other community detection general parameters, please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
-    "The Leiden algorithm is an improvement of the Louvain algorithm. The Leiden algorithm consists of three phases: (1) local moving of nodes, (2) refinement of the partition (3) aggregation of the network based on the refined partition, using the non-refined partition to create an initial partition for the aggregate network." - cdlib
+    For other community detection general parameters,
+    please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
+    "The Leiden algorithm is an improvement of the Louvain algorithm.
+    The Leiden algorithm consists of three phases:
+    (1) local moving of nodes,
+    (2) refinement of the partition,
+    (3) aggregation of the network based on the refined partition,
+    using the non-refined partition to create an initial partition for the aggregate network." - cdlib
```

Comment:
Cite the original citation of leiden algorithm. 

"- cdlib" make the a clear sentence. All information should be clear and complete, not fragmented, lazy description. May be you add the cdlib inform to the begining and say something like: 

"Based on the cdlib package, the Leiden algorithm consists of three phases:"